src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -252,146 +252,48 @@ winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::Duplicate
-        duplicated->Foreground(source.Foreground());
-    }
-    if (source.HasBackground() || source.BackgroundOverrideSource() != nullptr)
-    {
-        duplicated->Background(source.Background());
-    }
-    if (source.HasSelectionBackground() || source.SelectionBackgroundOverrideSource() != nullptr)
-    {
-        duplicated->SelectionBackground(source.SelectionBackground());
-    }
-    if (source.HasCursorColor() || source.CursorColorOverrideSource() != nullptr)
-    {
-        duplicated->CursorColor(source.CursorColor());
-    }
-    if (source.HasHistorySize() || source.HistorySizeOverrideSource() != nullptr)
-    {
-        duplicated->HistorySize(source.HistorySize());
-    }
-    if (source.HasSnapOnInput() || source.SnapOnInputOverrideSource() != nullptr)
-    {
-        duplicated->SnapOnInput(source.SnapOnInput());
-    }
-    if (source.HasAltGrAliasing() || source.AltGrAliasingOverrideSource() != nullptr)
-    {
-        duplicated->AltGrAliasing(source.AltGrAliasing());
-    }
-    if (source.HasCursorShape() || source.CursorShapeOverrideSource() != nullptr)
-    {
-        duplicated->CursorShape(source.CursorShape());
-    }
-    if (source.HasCursorHeight() || source.CursorHeightOverrideSource() != nullptr)
-    {
-        duplicated->CursorHeight(source.CursorHeight());
-    }
-    if (source.HasBellStyle() || source.BellStyleOverrideSource() != nullptr)
-    {
-        duplicated->BellStyle(source.BellStyle());
-    }
-    if (source.HasPixelShaderPath() || source.PixelShaderPathOverrideSource() != nullptr)
-    {
-        duplicated->PixelShaderPath(source.PixelShaderPath());
-    }
-    if (source.HasBackgroundImageAlignment() || source.BackgroundImageAlignmentOverrideSource() != nullptr)
-    {
-        duplicated->BackgroundImageAlignment(source.BackgroundImageAlignment());
-    }
+    #define DUPLICATE_MACRO(settingName)                                                                                                                            \
+    if (source.Has##settingName() ||                                                                                                                                \
+       (source.##settingName##OverrideSource() != nullptr && source.##settingName##OverrideSource().Origin() != OriginTag::UserDefaults))                           \
+    {                                                                                                                                                               \
+        duplicated->##settingName(source.##settingName());                                                                                                          \
+    }
+
+    DUPLICATE_MACRO(Hidden);
+    DUPLICATE_MACRO(Icon);
+    DUPLICATE_MACRO(CloseOnExit);
+    DUPLICATE_MACRO(TabTitle);
+    DUPLICATE_MACRO(TabColor);
+    DUPLICATE_MACRO(SuppressApplicationTitle);
+    DUPLICATE_MACRO(UseAcrylic);
+    DUPLICATE_MACRO(AcrylicOpacity);
+    DUPLICATE_MACRO(ScrollState);

```

Comment:
for @zadjii-msft or @carlos-zamora to re-review -- this is supposed to duplicate all settings that are _not_ set in BaseLayer -- it will promote defaults.json, Generated, Fragment or User-level settings into the new profile block, which I believe is correct.